### PR TITLE
zsteg: init at 0.2.2

### PR DESCRIPTION
**Pull request by applePrincess**
###### Motivation for this change
Add zsteg package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by buckley310** [comment]
This builds and ran fine for me against a steg file I had lying around.
Probably should squash the changes down to just the two commits though (maintainers list and then package itself), and change the commit message to match the PR title.

**Comment by buckley310** [comment]
All looks good after that push.

**Comment by buckley310** [comment]
`all-packages` is self-described as "roughly sorted by alphabet", although zsteg seems at home next to steghide, it might make sense to move it. Come to think of it, I'm not sure why steghide ended up in the `DEVELOPMENT / LIBRARIES` section anyway, I think its just a cli utility.

**Comment by applePrincess** [comment]
In short, I'm confused.

In all-packages.nix,

```
/* The top-level package collection of nixpkgs.
 * It is sorted by categories corresponding to the folder names
 * in the /pkgs folder. Inside the categories packages are roughly
 * sorted by alphabet, but strict sorting has been long lost due
 * to merges. Please use the full-text search of your editor. ;)
```

To my understanding, sort by category (approx. folder) first, then sort by alphabet.
But the actual fact is that, sorted by either proprely (?? to my understanding), alphabet only or relevalence.
I think, one day, we need a branch to solve this.

**Comment by buckley310** [comment]
If it were me, I would insert it alphabetically into the `APPLICATIONS` section.

**Comment by applePrincess** [comment]
Thank you for guiding me.

**Comment by applePrincess** [comment]
Is this PR couldn't get merged because of checks failure? 
The file failing is automatically generated by bundix, so this can be either a bug(??) in bundix or add exception to editorconfig.

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -686,6 +686,16 @@
     githubId = 1078530;
     name = "Alexandre Peyroux";
   };
+  applePrincess = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:689]: Please put this into a separate commit with the message ``maintainers: add applePrincess``.

File: pkgs/tools/security/zsteg/Gemfile [hunk 2]
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
+gem 'zsteg'
+
```
Comment (SuperSandro2000) [inline pkgs/tools/security/zsteg/Gemfile:3]: One new line to much.

File: pkgs/tools/security/zsteg/gemset.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    groups = ["default"];
+    platforms = [];
+    source = {
+      remotes = ["https://rubygems.org"];
+      sha256 = "1mwajlsgs27449n2yf2f9hz8g46qv9bz9f58i9cz1jg58spvpxpk";
+      type = "gem";
+    };
+    version = "0.2.2";
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/tools/security/zsteg/gemset.nix:44]: Missing final new line.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -26363,6 +26363,8 @@ in
 
   zscroll = callPackage ../applications/misc/zscroll {};
 
+  zsteg = callPackage ../tools/security/zsteg {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26366]:
```suggestion
  zsteg = callPackage ../tools/security/zsteg { };
```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -26377,6 +26379,7 @@ in
     guiModule = "zest";
   };
 
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26382]: Please revert that.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/security/zsteg/gemset.nix (hunk 3 above)
Comment (applePrincess) [reply pkgs/tools/security/zsteg/gemset.nix:44]: This file is what I mentioned above. This is generad by running ```$(nix-build '<nixpkgs>' -A bundix --no-out-link)/bin/bundix --magic``` which is stated by [Ruby section of Nixpkgs manual](https://nixos.org/manual/nixpkgs/stable/#sec-language-ruby). To me, changing machine generated files is a hack.

Comment (applePrincess) [reply pkgs/tools/security/zsteg/gemset.nix]: OK now I see trailing newline issue on [nix-community/bundix](https://github.com/nix-community/bundix) repo, so will soon be fixed, for this time I added manually.

**Comment by SuperSandro2000** [comment]
@ofborg eval
